staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -677,6 +683,29 @@ func (a *Admission) exemptRuntimeClass(runtimeClass *string) bool {
 	// TODO: consider optimizing to O(1) lookup
 	return containsString(*runtimeClass, a.Configuration.Exemptions.RuntimeClasses)
 }
+
+// Filter and prioritise pods based on runtimeclass and uniqueness of the controller respectively for evaluation
```

Comment:
runtimeclass isn't considered inside the function as documented

It would make sense to move the `if a.exemptRuntimeClass(pod.Spec.RuntimeClassName) {` check into `prioritizePods()` and skip accumulating them there before truncating to namespaceMaxPodsToCheck

That way, we only truncate the list of pods we would have actually wanted to check

Reply:
nit: prioritise → prioritize (change everywhere)

Reply:
add unit tests exercising exempt runtimeclass pods and replicated pods exceeding namespaceMaxPodsToCheck (can set namespaceMaxPodsToCheck low in the unit test to make testing easier), and ensure other pods are 

something like namespaceMaxPodsToCheck=3, with a podlister that returns:
- exemptruntimeclasspod1
- exemptruntimeclasspod2
- exemptruntimeclasspod3
- replicatedpod1
- replicatedpod2
- replicatedpod3
- uniquepod1
- uniquepod2
- uniquepod3

there should be a warning about only checking 3 of 6 non-exempt pods in the namespace, and the pods that should be checked are `replicatedpod1`, `uniquepod1`, and `uniquepod3`